core/utils/decimal.go

```diff
@@ -56,6 +66,17 @@ func ToDecimal(input interface{}) (decimal.Decimal, error) {
 	}
 }
 
+// hexStringToDecimal takes in a hex string, and returns (decimal.Decimal, bool), bool indicates success status
+func hexStringToDecimal(hexString string) (decimal.Decimal, bool) {
+	hexString = strings.TrimPrefix(hexString, "0x")
+	n := new(big.Int)
+	_, success := n.SetString(hexString, 16)
+	if !success {
+		return decimal.Zero, false
```

Comment:
I attempted to change this to:
`return decimal.Decimal{}, false`
But it broke an e2e test.

Worth an investigation, if this to be merged in.